images.go

```diff
@@ -742,7 +745,28 @@ func (r *imageStore) Create(id string, names []string, layer, metadata string, c
 		list := r.bydigest[digest]
 		r.bydigest[digest] = append(list, image)
 	}
+	defer func() {
+		if err != nil {
+			// now that the in-memory structures know about the new
+			// record, we can use regular Delete() to clean up if
+			// anything breaks from here on out
+			if e := r.Delete(image.ID); e != nil && !errors.Is(e, os.ErrNotExist) {
```

Comment:
The error from `Delete` is `ErrImageUnknown`, and I can’t see how it could fail with `ErrNotExist` (`RemoveAll` should not return that).

Reply:
Right, removing that `errors.Is()` call.